public/modules/custom/helfi_rekry_structured_data/helfi_rekry_structured_data.module

```diff
@@ -0,0 +1,106 @@
+<?php
+
+/**
+ * @file
+ * Functions for HELfi Rekry Structured data module.
+ */
+
+declare(strict_types = 1);
+
+use Drupal\Core\Entity\ContentEntityInterface;
+
+/**
+* Implements hook_page_attachments().
+*/
+function helfi_rekry_structured_data_page_attachments(array &$attachments) {
+
+  // Get current entity and entity version.
+  $entity_matcher = \Drupal::service('helfi_platform_config.entity_version_matcher')->getType();
+
+  /** @var \Drupal\Core\Entity\ContentEntityInterface $entity */
+  $entity = $entity_matcher['entity'];
+
+  // No need to continue if current entity doesn't have hero set.
+  if (
+    !$entity instanceof ContentEntityInterface ||
+    $entity->getType() !== 'job_listing'
+  ) {
+    return;
+  }
+
+  $publication_starts_datetime = strtotime($entity->get('field_publication_starts')->value);
+  $publication_starts_formatted = date('Y-m-d', $publication_starts_datetime);
```

Comment:
Use `date.formatter` service to format date.

Reply:
Done.